### ISSUE TITLE
Update installation instructions for Stampede2

### ISSUE THREAD

**Issue by gassmoeller** [geodynamics/aspect#1838]
We have installation instructions for Stampede in our /doc/install/ directory, but as far as I got from the TACC website (see also #1271 and #562) Stampede was integrated into a new system (Stampede2). @hlokavarapu you installed aspect on Stampede2, did you use candi for that? Did you had to modify anything? And could you maybe take a look at the instructions in /doc/install and see if they need to be updated?

**Comment by naliboff** [comment]
The attached file is a collection of notes from my installing ASPECT on Stampede2:
[Stampede2_ASPECT_Installation_Instructions.txt](https://github.com/geodynamics/aspect/files/1981709/Stampede2_ASPECT_Installation_Instructions.txt)

 I'll convert these to more general instructions on the ASPECT wiki, but first I thought it would be helpful to review the steps and see if everything makes sense. 

A number of things could likely be changed, including reducing the number of loaded modules, using deal.II's provided BOOST package (multiple warnings related to BOOST when compiling), etc.

Given that the configuration directions are quite lengthy, it seems like the best approach may be to provide an installation script? I have not tried installing with candi yet, but that would certainly simplify things as well (as would having pre-built p4est and Trilinos modules).

**Comment by bangerth** [comment]
When I started to write down my own installation instructions (for my own use), I learned that it is helpful to replace things like "Make sure following modules are loaded" and "Put tarball and shell script in a folder called p4est" by actual commands because it reduces the potential for mistakes when going down the list. I also try to avoid listing concrete directory names because they need to be edited when I give these instructions to other -- I use `$WORK` etc instead. 

So, the way I now typically do this is to write down complete scripts:
```
  # Make sure following modules are loaded
  module load gcc/7.1.0 impi/17.0.3 mkl/17.0.4 python/2.7.13 boost/1.64 swig/3.0.12 cmake/3.8.2 \
     phdf5/1.8.16 parallel-netcdf/4.3.3.1

  # Create a build directory for everything and go there
  cd $WORK
  mkdir bin
  cd bin
  mkdir packages
  cd packages

  ### p4est (1.1) Installation
  # 1. Download source code tarball (p4est-1.1.tar.gz) and shell script (p4est-setup.sh)
  wget https://www.dealii.org/developer/external-libs/p4est-setup.sh
  wget http://p4est.github.io/release/p4est-1.1.tar.gz

  ...
```
Written this way, the whole shebang can then simply be copy-pasted into a shell window, by everyone.

**Comment by naliboff** [comment]
@bangerth - Thanks for the tip! The instructions are certainly much cleaner now.

**Comment by naliboff** [comment]
I've added a page on the wiki for [ installing ASPECT on Stampede2](https://github.com/geodynamics/aspect/wiki/Compiling-and-Running-ASPECT-on-TACC-Stampede2). 

This will need to be cleaned up and expanded to include instructions for a manual (w/out candi) installation and submitting jobs, but the bulk of the info is now there.

@gassmoeller - Feel free to add your name to the authors as you kindly tested the candi installation first.

@jperryhouts @dsarahstamps - Likewise, feel free to add your names to the author list as I copied a good deal of text from instructions for the original Stampede (you are currently referenced). I also added a note to the old instructions that Stampede1 is currently out of service.

All - feel free to modify the current text and let me know if you have any questions/comments.
